plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/aps/system/services/resource/ResourcesService.java

```diff
@@ -0,0 +1,336 @@
+        }
+    }
+
+    public void deleteAsset(String resourceId) {
+        try {
+            ResourceInterface resource = resourceManager.loadResource(resourceId);
+            if (resource == null) {
+                throw new RestServerError(String.format("Resource %s not found", resourceId), null);
+            }
+            resourceManager.deleteResource(resource);
+        } catch (ApsSystemException e) {
+            throw new RestServerError("Error deleting image resource", e);
+        }
+    }
+
+    public AssetDto editAsset(String resourceId, MultipartFile file, String description, String group, List<String> categories) {
+        try {
+            ResourceInterface resource = resourceManager.loadResource(resourceId);
+            if (resource == null) {
+                throw new RestServerError(String.format("Resource %s not found", resourceId), null);
+            }
+
+            BaseResourceDataBean resourceFile = new BaseResourceDataBean();
+            resourceFile.setResourceType(resource.getType());
+            resourceFile.setResourceId(resourceId);
+            resourceFile.setMetadata(resource.getMetadata());
+
+
+            if (file != null) {
+                validateMimeType(resource.getType(), file.getContentType());
+
+                resourceFile.setInputStream(file.getInputStream());
+                resourceFile.setFileSize(file.getBytes().length / 1000);
+                resourceFile.setFileName(file.getOriginalFilename());
+                resourceFile.setMimeType(file.getContentType());
+                resourceFile.setDescr(file.getOriginalFilename());
+
+                if ((description == null || description.trim().length() == 0 ) && resource.getDescription().equals(resource.getMasterFileName())) {
+                    description = resourceFile.getFileName();
+                }
+            }
+
+            if (description != null && !description.trim().isEmpty()) {
+                resourceFile.setDescr(description.trim());
+            } else {
+                resourceFile.setDescr(resource.getDescription());
+            }
+
+            if (group != null && !group.trim().isEmpty()) {
+                resourceFile.setMainGroup(group.trim());
+            } else {
+                resourceFile.setMainGroup(resource.getMainGroup());
+            }
+
+            if (categories != null && categories.size() > 0) {
+                resourceFile.setCategories(categories.stream()
+                        .map(code -> categoryManager.getCategory(code)).collect(Collectors.toList()));
+            } else {
+                resourceFile.setCategories(resource.getCategories());
+            }
+
+            resourceManager.updateResource(resourceFile);
+            return convertResourceToDto(resource.getType(), resourceManager.loadResource(resourceId));
+        } catch (ApsSystemException e) {
+            throw new RestServerError("Error updating image resource", e);
```

Comment:
Should this get logged here?

Reply:
It is an unresolveable internal exception that shouldn't occur.
So throwing it wrapped in a known type and letting the generic exception handler log it, is the appropriate way to handle these kind of errors, in my opinion.

Reply:
They are being logged inside the `ResourceManager` before throwing the `ApsSystemException`.